src/auth/models.py

```diff
@@ -1,35 +1,42 @@
-from sqlalchemy import Column, DateTime, Integer, String
-# from sqlalchemy.orm import relationship
-
+from constants import COMPUTING_ID_LEN, SESSION_ID_LEN
 from database import Base
+from sqlalchemy import Column, DateTime, ForeignKey, String
+from sqlalchemy.orm import relationship
 
 
 class UserSession(Base):
     __tablename__ = "user_session"
 
     # note: a primary key is required for every database table
-    id = Column(Integer, primary_key=True)
-
-    issue_time = Column(DateTime, nullable=False)
-    session_id = Column(String(512), nullable=False)  # the space needed to store 256 bytes in base64
     computing_id = Column(
-        String(32), nullable=False
+        String(COMPUTING_ID_LEN), nullable=False, primary_key=True, unique=True
     )  # technically a max of 8 digits https://www.sfu.ca/computing/about/support/tips/sfu-userid.html
-    # TODO: link to the user's table entry & remove the computing_id column (it already exists in the User table)
+
+    issue_time = Column(DateTime, nullable=False)
+    session_id = Column(
+        String(SESSION_ID_LEN), nullable=False, unique=True
+    )  # the space needed to store 256 bytes in base64
+
+    site_user = relationship("SiteUser")
 
 
-class User(Base):
+class SiteUser(Base):
     # user is a reserved word in postgres
     # see: https://stackoverflow.com/questions/22256124/cannot-create-a-database-table-named-user-in-postgresql
     __tablename__ = "site_user"
 
     # note: a primary key is required for every database table
-    id = Column(Integer, primary_key=True)
+    computing_id = Column(
+        String(COMPUTING_ID_LEN),
+        ForeignKey("user_session.computing_id"),
+        nullable=False,
+        primary_key=True,
+        unique=True,
+    )  # technically a max of 8 digits https://www.sfu.ca/computing/about/support/tips/sfu-userid.html
+
+    officer_term = relationship("OfficerTerm", back_populates="site_user")
```

Comment:
are relationships nullable by default? Just checking

Reply:
So, some site users may be officers, and officer_term is their *current* officer term? + officer_info is extra information about them?

Reply:
relationships don't actually occupy any space on the disk, they're purely an sqlachemy concept for accessing related data. They get implemented by a join. If there's no item, then it will just be `None` (or `[]` if one to many)

Reply:
here's how I break it down:
- each site user has had some number of past exec terms. For each exec term, all their info might change. They might want a new paragraph, nickname, courses, etc.
- if a user is an exec, they upload some information about themselves that we want the most up to date version of. We don't care about old phone numbers, discord names, etc.